### PR TITLE
Terraform 2

### PR DESCRIPTION
**Pull request by lovejoi**
Я добавил так же свои переменные чтобы легче было расследовать так же вот ощибка при terraform apply

oleg@Oleg-DevOps:~/Training/Otus/lesson_of_terraform_v2/lovejoi_infra/terraform$ terraform apply
yandex_vpc_network.app-network: Refreshing state... [id=enp35rter3smc75c39vs]
yandex_vpc_subnet.app-subnet: Refreshing state... [id=e9bll19rii4t89c15u9t]

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # yandex_compute_instance.app will be created
  + resource "yandex_compute_instance" "app" {
      + created_at                = (known after apply)
      + folder_id                 = (known after apply)
      + fqdn                      = (known after apply)
      + hostname                  = (known after apply)
      + id                        = (known after apply)
      + labels                    = {
          + "tags" = "reddit-app"
        }
      + metadata                  = {
          + "ssh-keys" = <<~EOT
                ubuntu:ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDoRxvni8GeNUcBdpboYex5LjXfeGPmwbW7VkwIRsTmD6GLzE2cXdEB23cn1TFitHH/GODZMrGK9Nt+sSwKvseLhioEjKcBPokOc2K3xA6jXTa0RCM7r6p2b1d87Iq0H64TuBGZNn08tZ00FmyF6BOZzHNySHCGxlYmybE1MhTbyx3iUgMtyr0DqY1iQw15qLYNhU9y0yHHGyt5ul4MCkU2JBD1FnyL47dY2hTEvP6tpJNXLIi/sQZ8Z+EEOb5HpTzMkC2gqToxHeR5OPOoiLVqTqTza75taGjciwnek3KSuwHoUDzIlajcQqdquPmdh8AC4URRHbEiUz6HAFylv8br oleg@Oleg-DevOps
            EOT
        }
      + name                      = "reddit-app"
      + network_acceleration_type = "standard"
      + platform_id               = "standard-v1"
      + service_account_id        = (known after apply)
      + status                    = (known after apply)
      + zone                      = (known after apply)

      + boot_disk {
          + auto_delete = true
          + device_name = (known after apply)
          + disk_id     = (known after apply)
          + mode        = (known after apply)

          + initialize_params {
              + description = (known after apply)
              + image_id    = "reddit-app-base"
              + name        = (known after apply)
              + size        = (known after apply)
              + snapshot_id = (known after apply)
              + type        = "network-hdd"
            }
        }

      + network_interface {
          + index          = (known after apply)
          + ip_address     = (known after apply)
          + ipv4           = true
          + ipv6           = (known after apply)
          + ipv6_address   = (known after apply)
          + mac_address    = (known after apply)
          + nat            = true
          + nat_ip_address = (known after apply)
          + nat_ip_version = (known after apply)
          + subnet_id      = "e9bll19rii4t89c15u9t"
        }

      + resources {
          + core_fraction = 100
          + cores         = 1
          + memory        = 2
        }

      + scheduling_policy {
          + preemptible = (known after apply)
        }
    }

  # yandex_compute_instance.db will be created
  + resource "yandex_compute_instance" "db" {
      + created_at                = (known after apply)
      + folder_id                 = (known after apply)
      + fqdn                      = (known after apply)
      + hostname                  = (known after apply)
      + id                        = (known after apply)
      + labels                    = {
          + "tags" = "reddit-db"
        }
      + metadata                  = {
          + "ssh-keys" = <<~EOT
                ubuntu:ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDoRxvni8GeNUcBdpboYex5LjXfeGPmwbW7VkwIRsTmD6GLzE2cXdEB23cn1TFitHH/GODZMrGK9Nt+sSwKvseLhioEjKcBPokOc2K3xA6jXTa0RCM7r6p2b1d87Iq0H64TuBGZNn08tZ00FmyF6BOZzHNySHCGxlYmybE1MhTbyx3iUgMtyr0DqY1iQw15qLYNhU9y0yHHGyt5ul4MCkU2JBD1FnyL47dY2hTEvP6tpJNXLIi/sQZ8Z+EEOb5HpTzMkC2gqToxHeR5OPOoiLVqTqTza75taGjciwnek3KSuwHoUDzIlajcQqdquPmdh8AC4URRHbEiUz6HAFylv8br oleg@Oleg-DevOps
            EOT
        }
      + name                      = "reddit-db"
      + network_acceleration_type = "standard"
      + platform_id               = "standard-v1"
      + service_account_id        = (known after apply)
      + status                    = (known after apply)
      + zone                      = (known after apply)

      + boot_disk {
          + auto_delete = true
          + device_name = (known after apply)
          + disk_id     = (known after apply)
          + mode        = (known after apply)

          + initialize_params {
              + description = (known after apply)
              + image_id    = "reddit-db-base"
              + name        = (known after apply)
              + size        = (known after apply)
              + snapshot_id = (known after apply)
              + type        = "network-hdd"
            }
        }

      + network_interface {
          + index          = (known after apply)
          + ip_address     = (known after apply)
          + ipv4           = true
          + ipv6           = (known after apply)
          + ipv6_address   = (known after apply)
          + mac_address    = (known after apply)
          + nat            = true
          + nat_ip_address = (known after apply)
          + nat_ip_version = (known after apply)
          + subnet_id      = "e9bll19rii4t89c15u9t"
        }

      + resources {
          + core_fraction = 100
          + cores         = 1
          + memory        = 2
        }

      + scheduling_policy {
          + preemptible = (known after apply)
        }
    }

Plan: 2 to add, 0 to change, 0 to destroy.

Warning: Interpolation-only expressions are deprecated

  on vpc.tf line 8, in resource "yandex_vpc_subnet" "app-subnet":
   8:   network_id     = "${yandex_vpc_network.app-network.id}"

Terraform 0.11 and earlier required all non-constant expressions to be
provided via interpolation syntax, but this pattern is now deprecated. To
silence this warning, remove the "${ sequence from the start and the }"
sequence from the end of this expression, leaving just the inner expression.

Template interpolation syntax is still used to construct strings from
expressions when the template includes multiple interpolation sequences or a
mixture of literal strings and interpolations. This deprecation applies only
to templates that consist entirely of a single interpolation sequence.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

yandex_compute_instance.app: Creating...
yandex_compute_instance.db: Creating...

Error: Error create 'boot_disk' object of api request: Error on retrieve image properties: server-request-id = 66cbebb1-5bee-b249-aa77-131428ed93a5 server-trace-id = 97d1afc71ac61472:4d623159e03af036:97d1afc71ac61472:1 client-request-id = a05555dd-2441-4594-84a8-7dbed6a4880d client-trace-id = 8e534f20-bf84-48b1-98ae-0264d6f69cd6 rpc error: code = InvalidArgument desc = invalid image id 'reddit-app-base'

  on app.tf line 1, in resource "yandex_compute_instance" "app":
   1: resource "yandex_compute_instance" "app" {

Error: Error create 'boot_disk' object of api request: Error on retrieve image properties: server-request-id = a914a157-7ebd-b6b1-a3d0-3b5590933adf server-trace-id = b76483193ee9a308:eb3b79d3aac89ebe:b76483193ee9a308:1 client-request-id = 4e217934-3dbb-4444-b47b-b34aa41b011a client-trace-id = 8e534f20-bf84-48b1-98ae-0264d6f69cd6 rpc error: code = InvalidArgument desc = invalid image id 'reddit-db-base'

  on db.tf line 1, in resource "yandex_compute_instance" "db":
   1: resource "yandex_compute_instance" "db" {

### CODE REVIEW

**Comment by ssi91** [comment]
@lovejoi Обрати внимание на текст ошибки ` invalid image id 'reddit-app-base` и `invalid image id 'reddit-db-base'`.

File: terraform/variables.tf [hunk 1]
```diff
@@ -24,4 +24,12 @@ description = "~/.ssh/ubuntu"
...
 description = "~/.ssh/ubuntu.pub"
 }
+variable app_disk_image {
+description = "Disk image for reddit app"
+default = "reddit-app-base"
+}
+variable db_disk_image {
+  description = "Disk image for reddit db"
+  default = "reddit-db-base"
+}
```
Comment (ssi91) [inline terraform/variables.tf]: Тут ты объявил переменные `app_disk_image` и `app_disk_image`. Они же потом используются в конфигах террафарма [Тут](https://github.com/Otus-DevOps-2020-05/lovejoi_infra/pull/6/files#diff-a3b15de7a9979b144ba2b4e99c7bf3a7R14) и [Тут](https://github.com/Otus-DevOps-2020-05/lovejoi_infra/pull/6/files#diff-9c4a4ef6fc1b5b0e3fd43d17c3946524R14)
Теперь тебе следует задать их в твоём `.tfvars`[тут](https://github.com/Otus-DevOps-2020-05/lovejoi_infra/pull/6/files#diff-799ff0192398b49ab4413c02a47e920d). Сейчас они у тебя не заданы, поэтому используются значения по-умолчанию. Туда нужно указать не имена образов, а их ID

File: terraform/vpc.tf [hunk 2]
```diff
@@ -0,0 +1,10 @@
+resource "yandex_vpc_network" "app-network" {
+  name = "app-network"
+}
+
+resource "yandex_vpc_subnet" "app-subnet" {
+  name           = "app-subnet"
+  zone           = "ru-central1-a"
+  network_id     = "${yandex_vpc_network.app-network.id}"
```
Comment (ssi91) [inline terraform/vpc.tf]: Тут вылазит предупреждение `Warning: Interpolation-only expressions are deprecated`. Можно заменить на просто `network_id     = yandex_vpc_network.app-network.id` :)

**Review by ssi91 — CHANGES_REQUESTED** [review]

**Comment by mrgreyves** [comment]
Здравствуйте! Для начала вам снова нужно свериться с методичкой, нехватает достаточно большого количества файлов. Далее будем решать проблемы с самим терраформом.

File: terraform/terraform.tfvars.example [hunk 3]
```diff
@@ -1,8 +1,10 @@
...
+token = "AgAAAABCS__nAATuwfzImpKBEUtwjBgIzE-7Kww"
+cloud_id = "b1gf6ugjqc3frqm0ijqb"
+folder_id = "b1g4l494cgnejfisqgv2"
+zone = "ru-central1-a"
+image_id = "fd8gd09kaus9gphrtuni"
+subnet_id = "e9bd45khoqmli5tn0859"
+private_key_path = "~/.ssh/ubuntu"
+public_key_path = "~/.ssh/ubuntu.pub"
+reddit-db-base = "fd8jurpvvqtbgm38c5nq"
+reddit-app-base = "fd8t7fe6ig657ofvp094"
```
Comment (ssi91) [inline terraform/terraform.tfvars.example]: Но ведь переменные называются [`db_disk_image`](https://github.com/Otus-DevOps-2020-05/lovejoi_infra/pull/6/files#diff-de972723020f8865a774d8baa9f34fe4R31) и [`app_disk_image`](https://github.com/Otus-DevOps-2020-05/lovejoi_infra/pull/6/files#diff-de972723020f8865a774d8baa9f34fe4R27) соответственно :)
Внимательней :)

**Review by ssi91 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**